provider/pkg/provider/helm_release.go

```diff
@@ -361,19 +359,24 @@ func (r *helmReleaseProvider) Check(ctx context.Context, req *pulumirpc.CheckReq
 			return nil, err
 		}
 
-		err = r.setComputedInputs(ctx, urn, new)
+		chart, err := r.helmBuild(ctx, urn, new)
 		if err != nil {
-			// setComputedInputs fails when the chart cannot be rendered, and we report it as a problem with the `chart` input.
 			failures = append(failures, &pulumirpc.CheckFailure{
 				Property: "chart",
 				Reason:   fmt.Sprintf("%v; check the chart name and repository configuration.", err),
 			})
+		} else {
+			new.Version = chart.Metadata.Version
```

Comment:
Let's add a quick comment to document why we want the new input's version to be updated with the "computed" version.